python_ta/checkers/unnecessary_indexing_checker.py

```diff
@@ -89,30 +89,32 @@ def _is_load_subscript(index_node: astroid.Name, for_node: astroid.For) -> bool:
     and the subscript item is being loaded from, e.g., s += iterable[index_node].
     """
     iterable = _iterable_if_range(for_node.iter)
-    # Name node is not inside Subscript node iterable[index].
-    if not (isinstance(index_node.parent, astroid.Index) and isinstance(index_node.parent.parent, astroid.Subscript)
-            and isinstance(index_node.parent.parent.value, astroid.Name)
-            and index_node.parent.parent.value.name == iterable):
-        return False
-    # Use ctx attribute to find out what context the subscript is being used in.
-    subscript_node = index_node.parent.parent
-    return subscript_node.ctx == astroid.Load
+    # If the Name node is inside a Subscript node and the subscript is being used in a load context
+    if isinstance(index_node.parent, astroid.Subscript) \
+            and isinstance(index_node.parent.value, astroid.Name) \
+            and index_node.parent.value.name == iterable:
+        # Use ctx attribute to find out what context the subscript is being used in
+        subscript_node = index_node.parent
+        return subscript_node.ctx == astroid.Load
+
+    # If Name node is not inside a Subscript node iterable[Name]
+    return False
 
 
-def _is_redundant(index_node: Union[astroid.AssignName, astroid.Name], for_node: astroid.For) -> bool:
+def _is_redundant(index_node: Union[astroid.AssignName, astroid.Name],
+                  for_node: astroid.For) -> bool:
     """Return whether or not <index_node> is redundant in <for_node>.
 
     The lookup method is used in case the original loop variable is shadowed
     in the for loop's body.
     """
-    if isinstance(index_node, astroid.AssignName):
```

Comment:
It looks like you've removed this branch, but this loses existing behaviour, which is not good. Instead you should be attempting to fix the runtime error directly.

You should create test cases to explore what this code is doing. `AssignName` nodes can appear as the left-hand side of an `=`, as the target of a for loop, and as the target in a comprehension. Play around with each of these cases by filling in a code template that looks like

```python
for i in range(len(lst)):
    ...
```